reference/commands/install.rst

```diff
@@ -232,3 +232,58 @@ to only update those packages, which avoids the re-evaluation of the whole graph
 
 Note that the ``--update`` argument will look into all the remotes specified in the command for possible newer versions,
 and won't stop at the first newer one found.
+
+
+Build modes
+-----------
+
+The ``conan install --build=<mode>`` argument controls the behavior regarding building packages from source. 
+The default behavior is failing if there are no existing binaries, with the "missing binary" error message,
+except for packages that define a ``build_policy = "missing"`` policy, but this can be changed with the 
+``--build`` argument.
+
+The possible values are:
+
+.. code-block:: bash
+
+    --build=never      Disallow build for all packages, use binary packages or fail if a binary
+                       package is not found, it cannot be combined with other '--build' options.
+    --build=missing    Build packages from source whose binary package is not found.
+    --build=cascade    Build packages from source that have at least one dependency being built from
+                       source.
+    --build=[pattern]  Build packages from source whose package reference matches the pattern. The
+                       pattern uses 'fnmatch' style wildcards, so '--build="*"' will build everything
+                       from source.
```

Comment:
Maybe we need to clarify (Or warn in the client?) the fact that `--build=zlib` _won't_ build zlib, as the pattern is checked against name/version

Reply:
Done.
Also added new section in ``conan create`` too, to clarify some behavior.